userspace/libscap/scap.c

```diff
@@ -431,6 +431,87 @@ scap_t* scap_open_udig_int(char *error, int32_t *rc,
 }
 #endif // !defined(HAS_CAPTURE) || defined(CYGWING_AGENT)
 
+#ifdef HAS_ENGINE_GVISOR
+scap_t* scap_open_gvisor_int(char *error, int32_t *rc, scap_open_args *args)
+{
+	scap_t* handle = NULL;
+
+	//
+	// Allocate the handle
+	//
+	handle = (scap_t*) calloc(1, sizeof(scap_t));
+	if(!handle)
+	{
+		snprintf(error, SCAP_LASTERR_SIZE, "error allocating the scap_t structure");
+		*rc = SCAP_FAILURE;
+		return NULL;
+	}
+
+	handle->m_vtable = &scap_gvisor_engine;
+	handle->m_engine.m_handle = handle->m_vtable->alloc_handle(handle, handle->m_lasterr);
+	if(!handle->m_engine.m_handle)
+	{
+		snprintf(error, SCAP_LASTERR_SIZE, "error allocating the engine structure");
+		free(handle);
+		return NULL;
+	}
+
+	*rc = handle->m_vtable->init(handle, args);
+	if(*rc != SCAP_SUCCESS)
+	{
+		scap_close(handle);
+		free(handle);
```

Comment:
Isn't this a potential double free? `handle` is already free'd by `scap_close`

Reply:
Yes! Great catch. `scap_udig` has the same issue I believe. Fixed here anyways :+1: